### PR TITLE
[CIVIS-8660] ENH response objects: add repr and pprint, fix lists for `.json()`

### PR DESCRIPTION
**Pull request by jacksonlee-civis**
This pull request updates `civis.response.Response`:

* Fix `.json()` where nested list objects weren't handled correctly.
* Add repr and pretty-printing protocols to improve user experience

Also added `civis --version` and fixed a small bug for `civis_logger` along the way.
 
---

- [x] (For Civis employees only) Reference to a relevant ticket in the pull request title
- [x] Changelog entry added to `CHANGELOG.md` at the repo's root level
- [x] Description of change in the pull request description
- [x] If applicable, unit tests have been added and/or updated
- [x] The CircleCI builds have all passed

### CODE REVIEW

File: docs/source/responses.rst [hunk 1]
```diff
@@ -1,6 +1,63 @@
 API Responses
 =============
 
+A Civis API call from ``client.<endpoint>.<method>`` returns a :class:`civis.response.Response` object
+(or a :class:`civis.response.PaginatedResponse` object, if ``<method>`` is a "list" call):
```
Comment (jacksonlee-civis) [inline docs/source/responses.rst:5]: Added these notes to document / encourage the use of `.json()` and call out the immutability of the new response objects.

This page in the Sphinx docs is not just for `civis.response.Response`. We'll come back to this page in a future PR to add and organize notes better for other classes and functions mentioned.

File: src/civis/response.py [hunk 2]
```diff
@@ -231,9 +268,20 @@ def __setstate__(self, state):
         """Set the state when unpickling, to avoid RecursionError."""
         self.__dict__ = state
 
-    def _replace(self, key, value):
-        """Only used within this repo; `key` assumed to be in snake_case."""
-        self._data_snake[key] = value
```
Comment (jacksonlee-civis) [inline src/civis/response.py:236]: Dropping this internal method and refactoring this codebase to use `.json()` instead

File: tests/test_response.py [hunk 3]
```diff
@@ -325,6 +342,66 @@ def test_repr(json_data, expected_repr):
...
+        ),
+        (
+            {
+                "foo": {ascii_lowercase[i]: i for i in range(3)},
+                "fooBar": {ascii_lowercase[i]: i for i in range(15)},
+            },
+            "Response(foo=(a=0, b=1, c=2),\n"
+            "         foo_bar=(a=0,\n"
+            "                  b=1,\n"
+            "                  c=2,\n"
```
Comment (jacksonlee-civis) [inline tests/test_response.py:384]: Another doc note: In a future PR, maybe somewhere in the Sphinx docs we could encourage people to use either IPython or Jupyter notebooks for syntax highlighting and nicer pretty prints like this (the regular Python interactive interpreter or directly calling `print(response)` would have this rendered in one single unwieldy long line).

File: docs/source/responses.rst (hunk 1 above)
Comment (jacksonlee-civis) [reply docs/source/responses.rst:5]: The doc build has finished for a preview: https://civis-python--487.org.readthedocs.build/en/487/responses.html#api-responses

**Review by Drew-Pappas — APPROVED** [review]
LGTM! 

- Pulled in branch and checked for instances of `_replace()`
- Reviewed sphinx preview
- Ran tests

**Pull request merged**